src/main/java/com/programmers/java/Calculator.java

```diff
@@ -0,0 +1,99 @@
+        index = 0;
+        String postExp = toPostfix(expression);
+        index = 0;
+        BigDecimal answer = postfixCalculator(postExp);
+        history.add(expression+" = "+answer);
+        return answer;
+    }
+
+    public static BigDecimal postfixCalculator(String postExp) throws Exception {
+        Stack<BigDecimal> numberStack = new Stack<>();
+        while (index < postExp.length()) {
+            char ch = postExp.charAt(index);
+            if (Character.isDigit(ch)) {
+                numberStack.push(new BigDecimal(tokenizer(postExp)));
+            } else if (isOperator(ch)) {
+                numberStack.push(performOperation(ch, numberStack.pop(), numberStack.pop()));
+            }
+            index++;
+        }
+        return numberStack.pop();
+    }
+
+    private static String toPostfix(String expression) throws Exception {
+        Stack<Character> operatorStack = new Stack<>();
+        StringBuilder intermediateBuilder = new StringBuilder();
+        while(index < expression.length()) {
+            char ch = expression.charAt(index);
+            if(Character.isDigit(ch)) {
+                intermediateBuilder.append(tokenizer(expression)+ " ");
+            } else if (isOperator(ch)) {
+                while(!operatorStack.isEmpty() && getPriority(operatorStack.peek()) >= getPriority(ch)) {
+                    intermediateBuilder.append(operatorStack.pop());
+                }
+                operatorStack.push(ch);
+            } else if( expression.charAt(index) == '(') {
+                operatorStack.push(ch);
+            } else if( expression.charAt(index) == ')') {
+                while (operatorStack.peek() != '(') {
+                    intermediateBuilder.append(operatorStack.pop());
+                }
+                operatorStack.pop();
+            } else {
+                throw new Exception();
+            }
+
+            index++;
+        }
+        while (!operatorStack.isEmpty()) {
+            intermediateBuilder.append(operatorStack.pop());
+        }
+        return intermediateBuilder.toString();
+    }
+
+    private static String tokenizer(String exp) {
+        StringBuilder decimalBuilder = new StringBuilder();
+        while (index < exp.length() && (Character.isDigit(exp.charAt(index)) || exp.charAt(index) == '.')) {
+            decimalBuilder.append(exp.charAt(index));
+            index++;
+        }
+        index--;
+        return decimalBuilder.toString();
+    }
+    private static BigDecimal performOperation(Character operator, BigDecimal popLast, BigDecimal popSecondLast) {
+        switch (operator) {
+            case '+':
+                return popSecondLast.add(popLast);
+            case '-':
+                return popSecondLast.subtract(popLast);
+            case '*':
+                return popSecondLast.multiply(popLast);
+            case '/':
+                return popSecondLast.divide(popLast);
+        }
+        return BigDecimal.ZERO;
+    }

```

Comment:
switch-case 문에서 default는 가급적 추가해주세요..!

```suggestion
        switch (operator) {
            case '+':
                return popSecondLast.add(popLast);
            case '-':
                return popSecondLast.subtract(popLast);
            case '*':
                return popSecondLast.multiply(popLast);
            case '/':
                return popSecondLast.divide(popLast);
            default:
                return BigDecimal.ZERO;
        }
```